### ISSUE TITLE
[template/i18n] doesn't work with control flow syntax

### ISSUE THREAD

**Issue by mattlewis92** [angular-eslint/angular-eslint#1640]
**Description and reproduction of the issue**

```JSON
{
  "rules": {
    "@angular-eslint/template/i18n": ["error", {
      checkId: true,
      checkAttributes: false,
      checkDuplicateId: false,
      ignoreTags: [':svg:title', ':svg:desc', ':svg:tspan'],
    }]
  }
}
```

```HTML
<div
  i18n="@@foo"
>
  Name: @if (name) {
  <span>{{ name }}</span>
  }
</div>
```

The i18n rule should walk up past the control flow syntax and see the i18n tag on the parent div element. Instead it reports on the span element that an `i18n` attribute is required.

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `17.1.0` |
| `@angular-eslint/template-parser`         | `17.1.0` |
| `@typescript-eslint/parser`               | `6.11.0` |
| `ESLint`                                  | `8.46.0` |
| `node`                                    | `18.17.1` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 17.0.3
Node: 18.17.1
Package Manager: pnpm 8.10.5
OS: darwin arm64

Angular: 17.0.4
... animations, common, compiler, compiler-cli, core, elements
... forms, language-service, localize, platform-browser
... platform-browser-dynamic, router, service-worker

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1700.1
@angular-devkit/build-angular   17.0.3
@angular-devkit/core            17.0.3
@angular-devkit/schematics      17.0.3
@angular/cdk                    17.0.1
@angular/cli                    17.0.3
@angular/google-maps            15.2.9
@schematics/angular             17.0.3
ng-packagr                      17.0.2
rxjs                            7.8.0
typescript                      5.2.2
webpack                         5.88.2
zone.js                         0.14.2
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.
